roles/splunk_search_head/tasks/search_head_clustering.yml

```diff
@@ -70,14 +70,28 @@
     - Restart the splunkd service
   no_log: "{{ hide_password }}"
 
+- name: Add new member to SHC
+  command: "{{ splunk.exec }} add shcluster-member -current_member_uri {{ cert_prefix }}://{{ groups['splunk_search_head_captain'][0] }}:{{ splunk.svc_port }} -auth {{ splunk.admin_user }}:{{ splunk.password }}"
```

Comment:
My only question here is whether or not ```groups['splunk_search_head_captain'][0]``` will always be defined when we execute this block.  It should, I think.

What kinds of configurations did you test these changes with?

Reply:
As I understand it, this would be populated from `SPLUNK_SEARCH_HEAD_CAPTAIN_URL` environment variable. According to [current docs](https://github.com/splunk/splunk-ansible/blob/develop/docs/ADVANCED.md), this is required to enable SHC.

I tested by spinning up a new cluster, killing off and restarting various nodes (including captain), expanding cluster, killing off and restarting expanded nodes.